config/src/config/mod.rs

```diff
@@ -210,6 +210,54 @@ impl fmt::Display for RoleType {
 #[error("Invalid node role: {0}")]
 pub struct ParseRoleError(String);
 
+#[derive(Clone, Copy, Deserialize, Eq, PartialEq, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum NetworkType {
+    Validator,
+    FullNode,
+}
```

Comment:
is this type a projection on `NetworkId`? in other words, every `NetworkId` is one of these `NetworkType`s? it would make sense to me to add `NetworkId::network_type(&self) -> NetworkType` then.

as for the usage, it does seem useful for distinguishing e.g. a node with role=Validator and network_type=FullNode.

Reply:
Yeah, it is.  I can come back around and just remove the `NetworkType` mostly.  There's just some places where having a type is easier than dealing with converting to an `is_validator`